manim/mobject/coordinate_systems.py

```diff
@@ -731,3 +739,298 @@ def get_coordinate_labels(self, *numbers, **kwargs):
+
+            class PolarPlaneUnits(Scene):
+                def construct(self):
+                    self.add(polarplot_pi)
+                    self.wait(2)
+                    self.play(FadeOutAndShift(polarplot_pi, LEFT))
+                    self.play(FadeInFrom(polarplot_tau, RIGHT))
+                    self.wait(2)
+                    self.play(FadeOutAndShift(polarplot_tau, LEFT))
+                    self.play(FadeInFrom(polarplot_degrees, RIGHT))
+                    self.wait(2)
+                    self.play(FadeOutAndShift(polarplot_degrees, LEFT))
+                    self.play(FadeInFrom(polarplot_gradians, RIGHT))
+                    self.wait(2)
+
+    azimuth_offset : :class:`float`, optional
+        The angle offset of the azimuth labels, expressed in radians.
+
+    azimuth_direction : :class:`str`, optional
+        The direction of the azimuth labels.
+
+        - ``"CW"``: Clockwise.
+        - ``"CCW"``: Anti-clockwise.
+
+    azimuth_buff : :class:`int`, optional
+        The buffer for the azimuth labels.
+
+    azimuth_scale : :class:`int`, optional
+        The scale of the azimuth labels.
+
+    radius_config: :class:`dict`, optional
+        The axis config for the radius.
+
+    radius_max : :class:`float`, optional
+        The maximum value of the radius.
+    """
+
+    def __init__(
+        self,
+        color=BLUE,
+        azimuth_line_frequency=None,
+        radius_line_frequency=1,
+        azimuth_units="PI radians",
+        azimuth_offset=0,
+        azimuth_direction="CCW",
+        azimuth_buff=SMALL_BUFF,
+        azimuth_scale=0.5,
+        radius_config={"label_direction": DOWN + LEFT},
+        radius_max=None,
+        **kwargs,
+    ):
+        if azimuth_units in ["PI radians", "TAU radians", "degrees", "gradians", None]:
+            self.azimuth_units = azimuth_units
+        else:
+            raise ValueError(
+                "Invalid azimuth units. Expected one of: PI radians, TAU radians, degrees, gradians or None."
+            )
+        self.azimuth_frequency = (
+            {
+                "PI radians": (1 / 20),
+                "TAU radians": (1 / 20),
+                "degrees": (1 / 36),
+                "gradians": (1 / 40),
+                None: 1,
+            }[self.azimuth_units]
+            if azimuth_line_frequency is None
+            else azimuth_line_frequency
+        )
+        super().__init__(
+            color=color,
+            x_line_frequency=self.azimuth_frequency * TAU,
+            y_line_frequency=radius_line_frequency,
+            x_axis_config=radius_config,
+            x_min=None if (radius_max is None) else -radius_max,
+            x_max=radius_max,
+            y_min=-config["frame_x_radius"] if (radius_max is None) else -radius_max,
+            y_max=config["frame_x_radius"] if (radius_max is None) else radius_max,
+            **kwargs,
+        )
```

Comment:
```suggestion

        x_min = (-config["frame_x_radius"] if (radius_max is None) else -radius_max)
        x_max = (config["frame_x_radius"] if (radius_max is None) else radius_max)
        x_line_frequency = (self.azimuth_frequency * TAU)

        y_min = (-config["frame_y_radius"] if (radius_max is None) else -radius_max)
        y_max = (config["frame_y_radius"] if (radius_max is None) else radius_max)
        y_line_frequency = radius_line_frequency

        super().__init__(
            x_range=(x_min, x_max, x_line_frequency),
            y_range=(y_min, y_max, y_line_frequency),
            x_axis_config=radius_config,
            color=color,
            **kwargs,
        )

Reply:
Thanks! I can look at refactoring it to work without deprecated code for the next release. I've been told the whole inheritance  tree for `coordinate_systems` might be rewritten in the near future anyway, so it might be best to wait for that to happen and change things then.